frontend/ui/impl/src/test/java/com/virtuslab/gitmachete/frontend/ui/impl/GitMacheteErrorReportSubmitterTest.java

```diff
@@ -0,0 +1,76 @@
+package com.virtuslab.gitmachete.frontend.ui.impl;
+
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.never;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+
+import com.intellij.diagnostic.IdeaReportingEvent;
+import com.intellij.diagnostic.LogMessage;
+import com.intellij.openapi.diagnostic.IdeaLoggingEvent;
+import kr.pe.kwonnam.slf4jlambda.LambdaLogger;
+import lombok.val;
+import org.apache.commons.io.IOUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({GitMacheteErrorReportSubmitter.class})
+public class GitMacheteErrorReportSubmitterTest {
+
+  private String defaultErrorReport;
+  private String defaultErrorReportWithTemplate;
+
+  @Before
+  public void setUp() {
+    {
+      try {
+        defaultErrorReport = IOUtils.resourceToString("/error_report_uri.txt", StandardCharsets.UTF_8).strip();
+        defaultErrorReportWithTemplate = IOUtils.resourceToString("/error_report_uri_with_template.txt",
+            StandardCharsets.UTF_8).strip();
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  @Test
+  public void testGitHubIssueUriConstruction() throws Exception {
+    val events = new IdeaLoggingEvent[1];
+
+    val messageObject = Whitebox.newInstance(LogMessage.class);
+
+    events[0] = new IdeaReportingEvent(messageObject, /* message */ null, /* stacktrace */"", /* plugin */null);
+
+    val reportSubmitter = PowerMockito.spy(new GitMacheteErrorReportSubmitter());
+
+    PowerMockito
+        .stub(PowerMockito.method(GitMacheteErrorReportSubmitter.class, "getTemplateVariables", IdeaLoggingEvent[].class,
```

Comment:
How about just mocking e.g. `ApplicationInfo.getInstance().getFullApplicationName()` and `PluginManagerCore.getPlugin(...)` instead of full method of a tested object? would it be easily doable?

Reply:
Generally we should avoid mocking methods of the class that we're testing... it should be the **external world** that might be mocked instead (in our case, the IntelliJ-provided environment like `ApplicationInfo`)

Reply:
I thought about it, could be tricky but I will try it.

Reply:
Yeah, pls check... or at least limit the scope of mocked methods to as narrow as possible, sth like:

```java
private String getIdeApplicationName() {
  return ApplicationInfo.getInstance().getFullApplicationName();
}
```

so that major methods like `getTemplateVariables` remain **un**mocked (and hence truly covered with tests)